website/events/views.py

```diff
@@ -179,3 +181,30 @@ def dispatch(self, request, *args, **kwargs):
         except RegistrationError:
             pass
         return redirect(self.event)
+
+
+@method_decorator(login_required, name="dispatch")
+class MarkPresentView(View):
+    """A view that allows uses to mark their presence at an event using a secret token."""
+
+    def get(self, request, *args, **kwargs):
+        """Mark a user as present.
+
+        Checks if the url is correct, the event has not ended yet, and the user is registered.
+        """
+        event = get_object_or_404(Event, pk=kwargs["pk"])
+        if kwargs["token"] != event.mark_present_url_token:
+            messages.error(request, _("Invalid url."))
+        elif not request.member or not is_user_registered(request.member, event):
+            messages.error(request, _("You are not registered for this event."))
+        elif event.end < timezone.now():
+            messages.error(request, _("This event has already ended."))
```

Comment:
What if the event has not started yet?

Reply:
I'm not sure what to do there either. I can see it being used some time before events start (e.g. starting at 12:15 for lunch lectures), but there's really no sensible amount of time to place a limit at. 

Setting lower bound on the time doesn't matter at all anyway: the url is `secret` until the organizer decides to share it, and if someone is very early using the url, then forbidding it until the event starts still doesn't prevent them from using it as soon as the event starts. So I don't think there's really any reason to add a lower bound.